pyam/core.py

```diff
@@ -261,34 +261,43 @@ def metadata(self, meta, name=None):
             (by `['model', 'scenario']` index if possible)
         name: str
             category column name (if not given by meta pd.Series.name)
+        index: pd.MultiIndex
+            index to be used for setting metadata column
         """
-        if isinstance(meta, pd.Series) and \
-                set(['model', 'scenario']).issubset(meta.index.names):
-            meta.name = name or meta.name
+        # create pd.Series from index and meta if provided
+        _meta = meta if index is None else pd.Series(data=meta, index=index)
+
+        # append by index of pd.Series if possible
+        if isinstance(_meta, pd.Series) and \
+                set(['model', 'scenario']).issubset(_meta.index.names):
+            _meta.name = name or _meta.name
             # reduce index dimensions to model-scenario only
-            meta = (meta
-                    .reset_index()
-                    .loc[:, ['model', 'scenario', meta.name]]
-                    .set_index(['model', 'scenario'])
-                    )
+            _meta = (_meta
+                     .reset_index()
+                     .loc[:, ['model', 'scenario', _meta.name]]
+                     .set_index(['model', 'scenario'])
+                     )
             # raise error if index is not unique
-            if meta.index.duplicated().any():
+            if _meta.index.duplicated().any():
                 raise ValueError("non-unique ['model', 'scenario'] index!")
             # check if trying to add model-scenario index not existing in self
-            diff = meta.index.difference(self.meta.index)
+            diff = _meta.index.difference(self.meta.index)
```

Comment:
this should probably be combined with the below line as you do in 281

Reply:
reason for keeping `diff` explicitly is for using it in the ValueError message